src/Simplex/Messaging/Agent.hs

```diff
@@ -1912,6 +1935,18 @@ processSMPTransmission c@AgentClient {smpClients, subQ} (tSess@(_, srv, _), v, s
                               _ -> pure Nothing
                       _ -> prohibited >> ack
                   _ -> prohibited >> ack
+              updateConnVersion :: Connection c -> ConnData -> Version -> m (Connection c, ConnData)
+              updateConnVersion conn' cData' agentVersion = do
+                aVRange <- asks $ smpAgentVRange . config
+                case versionToRange agentVersion `compatibleVersion` aVRange of
+                  Just _
+                    | agentVersion > connAgentVersion -> do
+                      withStore' c $ \db -> setConnAgentVersion db connId agentVersion
+                      let cData'' = cData' {connAgentVersion = agentVersion} :: ConnData
```

Comment:
you actually need to use the result of compatibleVersion to update the version, you can't just use agentVersion as it can be higher than compatible.

```suggestion
                case versionToRange agentVersion `compatibleVersion` aVRange of
                  Just (Compatible v)
                    | v > connAgentVersion -> do
                      withStore' c $ \db -> setConnAgentVersion db connId v
                      let cData'' = cData' {connAgentVersion = v} :: ConnData
```

Reply:
e.g. if connection is set to 2, your supported range is 1-3 and you receive 4, you want to increase to 3 not to 4 - that's what compatibleVersion determines.

Now I am thinking that maybe it's incorrect to use compatibleVersion at all, as it won't actually return 3 in this case, and to make it do it you need to pass range (minVersion aVRange, agentVersion)?